src/nonblocking.jl

```diff
@@ -255,15 +255,20 @@ each request.
 # External links
 $(_doc_external("MPI_Waitall"))
 """
-function Waitall(reqs::RequestSet, statuses::Union{AbstractVector{Status},Nothing}=nothing)
-    n = length(reqs)
-    n == 0 && return nothing
-    @assert isnothing(statuses) || length(statuses) >= n
+function Waitall(count::Integer, reqs::AbstractVector{MPI_Request}, statuses::Union{AbstractVector{Status},Nothing}=nothing)
```

Comment:
Can you give an example of how you would use this? Where are you getting the `Vector{MPI_Request}` object from?

Reply:
Well, this comes from a `>20k` lines `Fortran` code which I am trying to map 1-to-1 to `julia` for strict comparison, before optimizing anything or rewriting in a `julian` way.

Vector of requests always have similar lengths and are allocated in an OO way somewhere else, and reused when synchronizing halo arrays, in a classic way multiple times per time-step.

Reply:
okay, that makes sense